src/Taxonomies/TermQueryBuilder.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Statamic\Eloquent\Taxonomies;
+
+use Illuminate\Support\Str;
+use Statamic\Contracts\Taxonomies\Term as TermContract;
+use Statamic\Facades\Collection;
+use Statamic\Facades\Site;
+use Statamic\Query\EloquentQueryBuilder;
+use Statamic\Taxonomies\TermCollection;
+
+class TermQueryBuilder extends EloquentQueryBuilder
+{
+    protected $collections = [];
+
+    protected $site = null;
+
+    protected $taxonomies = [];
+
+    protected $columns = [
+        'id', 'data', 'site', 'slug', 'uri', 'taxonomy', 'created_at', 'updated_at',
+    ];
+
+    protected function transform($items, $columns = [])
+    {
+        $site = $this->site;
+        if (! $site) {
+            $site = Site::default()->handle();
+        }
+
+        return TermCollection::make($items)->map(function ($model) use ($site) {
+            return app(TermContract::class)::fromModel($model)->in($site);
+        });
+    }
+
+    protected function column($column)
+    {
+        if (! is_string($column)) {
+            return $column;
+        }
+
+        if (! in_array($column, $this->columns)) {
+            $column = 'data->'.$column;
+        }
+
+        return $column;
+    }
+
+    public function where($column, $operator = null, $value = null, $boolean = 'and')
+    {
+        if ($column === 'site') {
+            $this->site = $operator;
+
+            return $this;
+        }
+
+        if (func_num_args() === 2) {
+            [$value, $operator] = [$operator, '='];
+        }
+
+        if (in_array($column, ['taxonomy', 'taxonomies'])) {
+            if (! $value) {
+                return $this;
+            }
+
+            if (! is_array($value)) {
+                $value = [$value];
+            }
+
+            $this->taxonomies = array_merge($this->taxonomies, $value);
+
+            return $this;
```

Comment:
Hi! I've been doing some testing, and noticed that term queries don't seem to `where` correctly on the taxonomies they belong to. It can be clearly reproduced when looking at the list of terms for a taxonomy in the CP - all listings always get all terms regardless. 
Did some digging, and I'm pretty sure this particular return statement (as well as the one on line 86) shouldn't be here? It prevents the function from ever reaching `parent::where(...)`, which it needs... right? Please correct me if I'm wrong here!

Reply:
Thanks for reporting - the issues are fixed here: https://github.com/statamic/eloquent-driver/pull/42/commits/46de559bf1fd27c9ca472f2a25b06434e20d3cb6